game-core/src/test/java/games/strategy/triplea/TripleATest.java

```diff
@@ -0,0 +1,59 @@
+package games.strategy.triplea;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.chat.Chat;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.framework.ServerGame;
+import games.strategy.engine.framework.startup.launcher.LaunchAction;
+import games.strategy.engine.framework.startup.ui.PlayerType;
+import games.strategy.engine.player.Player;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TripleATest {
+
+  private static final String DELEGATE_NAME_EDIT = "edit";
+
+  @Mock private LaunchAction launchAction;
+  @Mock private Chat chat;
+  @Mock private ServerGame serverGame;
+  @Mock private GameData gameData;
+
+  @Test
+  void testStartGameWhenServerGameStartedWithNewPlayersWithoutEditDelegate() {
+
+    final TripleA tripleA = new TripleA();
+    final Map<String, PlayerType> playerNames = new HashMap<>();
+    playerNames.put("first", PlayerType.HUMAN_PLAYER);
+    playerNames.put("second", PlayerType.WEAK_AI);
+    playerNames.put("third", PlayerType.PRO_AI);
+
+    when(serverGame.getData()).thenReturn(gameData);
+    when(gameData.getDelegate(DELEGATE_NAME_EDIT)).thenReturn(null);
+
+    final Set<Player> players = tripleA.newPlayers(playerNames);
+    assertThat(players, hasSize(3));
```

Comment:
It looks like TripleA has a factory method here. Do we expect side effects from `newPlayers` call? Are we simply checking that we can create 3 `Player` instances? 

It kinda seems like this is a test/check in of itself. What do you think to move this assertion and setup to its own test method?

Reply:
Yes it's not an interesting test, I thought I'll just check that `newPlayers` with an input of 3 names is expected to return 3 players. It should probably be moved to it's own test case,